overwatch/src/overwatch/handle.rs

```diff
@@ -120,25 +121,112 @@ where
         })
     }
 
-    /// Send a shutdown signal to the
-    /// [`OverwatchRunner`](crate::overwatch::OverwatchRunner)
-    pub async fn shutdown(&self) {
-        info!("Shutting down Overwatch");
+    /// Send a start signal to the specified service.
+    ///
+    /// # Errors
+    ///
+    /// If the start signal cannot be successfully delivered to the specified
+    /// service.
+    pub async fn start_service<Service>(&self) -> Result<(), ServiceError>
+    where
+        RuntimeServiceId: AsServiceId<Service>,
+    {
+        info!("Starting service with ID {}", RuntimeServiceId::SERVICE_ID);
+
+        let (sender, mut receiver) = tokio::sync::broadcast::channel(1);
+        self.send(OverwatchCommand::ServiceLifeCycle(
+            ServiceLifeCycleCommand {
+                service_id: RuntimeServiceId::SERVICE_ID,
+                msg: LifecycleMessage::Start(sender),
+            },
+        ))
+        .await
+        .map_err(|e| {
+            dbg!(e);
+            ServiceError::Start
+        })?;
+        receiver.recv().await.map_err(|e| {
```

Comment:
In `handle_service_service_lifecycle` of overwatch, we do the following:

```rust
ServiceLifeCycleCommand {
                service_id,
                msg: start_msg @ LifecycleMessage::Start(_),
            }
```

so we don't use the received `sender` in the lifecycle message, yet here we call `recv()` on the `receiver`. I don't see how and why this works. Can you explain it to me?

Reply:
Well, it is just a signal that the service started so it is "safe" to just do not wait for it. Now, why would we keep that signal if we do not use it. IMO we do not need to signal that. We have other things in place to know if a service is running or not (state watcher iirc).

Reply:
Yeah my question was why the `receiver.recv()` does not fail with a timeout if we never use the `sender` to send the signal. Or maybe it does in which case it would make sense since we don't send any signal.

Reply:
Does this somehow mean we could get rid of this information then? If the same sender is used in one place but not in the other? What happens if we were to use it also from here, then?

Reply:
Sorry, I'm still not following.
You're esentially saying we're not using the `sender`?
In that case the answer is: yes, we are. It's being forwarded to `OverwatchRunner` and later to `ServiceRunner`.
If that's not the case then I'll ask you to elaborate the question in a different manner 😅